### ISSUE TITLE
Am I understanding engergy function right by the following statement?

### ISSUE THREAD

**Issue by yongchanghao** [lifu-tu/ENGINE#1]
Is it saying that, the energy is equivalent to the cross entropy loss of a pretrained ```model_E``` where target side input replaced by O1(Y) weighted embedding, and target distribution is O2(Y_nograd)?

**Comment by lifu-tu** [comment]
It is close to what we have done. The energy in the paper is the sum of the negative log probabilities from the pretrained autoregressive model_E over all the positions. For each cross-entropy loss, the inference networks could get gradient from the two operators O1 and O2 (not just O1). 
In this paper, we just present one form of energy. Some other energy terms could be also used in the future.

**Comment by yongchanghao** [comment]
I see, thank you!
